components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -57,8 +57,22 @@ function Main() {
     siteInfo === null && // SiteInfo request has finished and this is a standalone conversation
     !context.isPremiumUser
 
+  const isLastTurnBraveSearchSERPSummary = () => {
+    const lastTurn = context.conversationHistory.at(-1);
+
+    if (!lastTurn?.events?.length) {
+      return false;
+    }
+
+    return lastTurn.events.some(
+      event => event.completionEvent?.fromBraveSearchSERP ?? false);
```

Comment:
nit:

Can events be undefined if `lastTurn` is defined? Intuitively I'd guess no but not really sure

in which case you could remove the second `?`


```suggestion
    if (!lastTurn?.events.length) {
      return false;
    }

    return lastTurn.events.some(
      event => event.completionEvent?.fromBraveSearchSERP ?? false);
```

optional: you could also collapse them


```suggestion
    return lastTurn?.events.some(
      event => event.completionEvent?.fromBraveSearchSERP) ?? false;
```

Reply:
I didn't want to make this assumption, and just want to check any possibilities in theory.

Reply:
Fair enough!